MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/MainActivity.java

```diff
@@ -581,4 +625,90 @@ protected void onSaveInstanceState(Bundle outState) {
     super.onSaveInstanceState(outState);
     outState.putInt("CURRENT_CATEGORY", currentCategory);
   }
+
+  private void checkForFirstTimeOpen() {
+    FirstTimeRunChecker firstTimeRunChecker = new FirstTimeRunChecker(this);
+    if (firstTimeRunChecker.firstEverOpen()) {
+      AnalyticsTracker.getInstance(getApplicationContext()).openedAppForFirstTime(
+        getResources().getBoolean(R.bool.isTablet), loggedIn);
+    }
+    firstTimeRunChecker.updateSharedPrefWithCurrentVersion();
+  }
+
+  private void buildSettingsDialog() {
+    AnalyticsTracker.getInstance(getApplicationContext()).trackEvent("Clicked on settings in nav drawer",
+      loggedIn);
+    LayoutInflater inflater = (LayoutInflater) getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+    View customView = inflater.inflate(R.layout.settings_dialog_layout, null);
+    analyticsOptOutSwitch = (Switch) customView.findViewById(R.id.analytics_opt_out_switch);
+    analyticsOptOutSwitch.setChecked(!AnalyticsTracker.getInstance(getApplicationContext()).isAnalyticsEnabled());
+    new AlertDialog.Builder(this)
```

Comment:
This method is too long. How about you have a new class called `SettingsDialogView` or similar where you have all this logic and instead of inner classes, you have that class implement the interfaces?